modules/islandora_collection_permissions/includes/handler.class.inc

```diff
@@ -191,20 +191,24 @@ class IslandoraCollectionPermissionsHandler {
     }
 
     $object = islandora_object_load($pid);
-    $related = array();
-
-    foreach (array('isMemberOfCollection', 'isMemberOf') as $pred) {
-      foreach ($object->relationships->get(FEDORA_RELS_EXT_URI, $pred) as $relation) {
-        $related[] = $relation['object']['value'];
+    if (is_object($object)) {
+      $related = array();
```

Comment:
This should probably stay outside of the `if` statement - and same with the `return` statement below. That way, if no related objects are found, this function will return an empty array instead of `NULL`. Otherwise this could break things if other code is relying on this function returning something that is countable.

Reply:
👍